### PR TITLE
Bug/issue 151/invalid trade offer creation

### LINKED ISSUE #151: Invalid trade offer creation
The `AddTradeForm` form in the `/profile` route is now invalid because of recent changes in the backend.

To stop preventing users from being **able to create new trade offers,** it must therefore be **updated** to allow them to again

### PR DESCRIPTION
**Pull request by TazDeCoder**
This PR closes #151

### CODE REVIEW

File: packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx [hunk 1]
```diff
@@ -40,22 +44,54 @@ export const AutoComplete = <T extends object>({
...
+    },
+    [onSelect, optionKeys.label, query],
   );
 
+  useEffect(() => {
+    const timeout = setTimeout(changedValueHandler, DEBOUNCE_DELAY);
+
+    return () => {
+      clearTimeout(timeout);
+    };
```
Comment (fmtabbara) [inline packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx:86]: can you use the useDebounce hook we have here?

**Review by fmtabbara — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx (hunk 1 above)
Comment (TazDeCoder) [reply packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx:86]: I've considered to use the hook, but I don't think it would work well with functions and this use case since functions are not **primitive values** so therefore it may not be ideal - wdyt?

Comment (fmtabbara) [reply packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx:86]: It's quite hard to follow the code. The useEffect will be called whenever the deps change. The deps change when `changedValueHandler` is called. `changedValueHandler` is called when then combo-box changes. So it seems like `changedValueHandler` is going to get called multiple times each time the combo box is changed.

Ideally you want to use `useDebounce` and pass in a value. `useDebounce` will give you back the value debounced. When then debounced value changes, you can use it inside a useEffect or call a callback prop with then value.

Comment (TazDeCoder) [reply packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx]: Changes made!
